src/puffin/src/puffin_manager/fs_puffin_manager/reader.rs

```diff
@@ -161,13 +163,24 @@ where
                     }
                 );
 
-                let reader = file.blob_reader(blob_meta)?;
+                let reader = accessor.reader(&puffin_file_name).await?;
                 let writer = writer_provider.writer(&file_meta.relative_path).await?;
-
-                let compression = blob_meta.compression_codec;
-                size += Self::handle_decompress(reader, writer, compression).await?;
+                let task = common_runtime::spawn_read(async move {
+                    let mut file = PuffinFileReader::new(reader);
+                    let reader = file.blob_reader(&blob_meta)?;
+                    let compression = blob_meta.compression_codec;
+                    let size = Self::handle_decompress(reader, writer, compression).await?;
+                    Ok(size)
+                });
+                tasks.push(task);
             }
 
+            let size = futures::future::try_join_all(tasks.into_iter())
+                .await
+                .into_iter()
+                .flatten()
+                .try_fold(0, |x, y| Ok(x + y?))?;
```

Comment:
non-blocking: Can we use sum?
```suggestion
                .await?
                .into_iter()
                .sum::<u64>();
```